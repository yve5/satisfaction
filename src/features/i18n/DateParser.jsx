import DateParser from './components/DateParser';
export default DateParser;
