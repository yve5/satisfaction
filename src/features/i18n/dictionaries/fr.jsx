export const I18N_DICTIONARY = {};
