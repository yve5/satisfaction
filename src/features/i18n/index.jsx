import I18n from './components/I18n';

export default I18n;
