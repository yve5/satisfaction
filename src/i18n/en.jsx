const I18N_DICTIONARY = {
  LANG: 'en',
  'Are you happy with your salary?': 'Are you happy with your salary?',
  'Are you satisfied with the interaction with people inside or outside the company?':
    'Are you satisfied with the interaction with people inside or outside the company?',
  'Confidence factor': 'Confidence factor',
  'Confidence factor calculation': 'Confidence factor calculation',
  'Do you actually enjoy doing the tasks of your job?':
    'Do you actually enjoy doing the tasks of your job?',
  'Do you appreciate the quality of the teamwork?':
    'Do you appreciate the quality of the teamwork?',
  'Do you have time for your personal projects?':
    'Do you have time for your personal projects?',
  'Do you like your workplace?': 'Do you like your workplace?',
  'Do your career prospects suit you?': 'Do your career prospects suit you?',
  'Do your colleagues agree with you?': 'Do your colleagues agree with you?',
  'Does doing this work bring you closer to your career goals?':
    'Does doing this work bring you closer to your career goals?',
  'Is your company doing well?': 'Is your company doing well?',
  Print: 'Print',
  Result: 'Result',
  'Source : Brett Nelson': 'Source : Brett Nelson',
  'The job is for you.': 'The job is for you.',
  'Think twice about taking the job, or leaving it right away.':
    'Think twice about taking the job, or leaving it right away.',
  'You got the perfect job!': 'You got the perfect job!',
  "You're on the right track to finding the perfect job.":
    "You're on the right track to finding the perfect job.",
};

export default I18N_DICTIONARY;
