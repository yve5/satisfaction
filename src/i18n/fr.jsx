const I18N_DICTIONARY = {
  LANG: 'fr',
  'Are you happy with your salary?': 'Votre salaire vous convient-il ?',
  'Are you satisfied with the interaction with people inside or outside the company?':
    'L’intéraction avec les personnes en ou hors de l’entreprise vous satisfait-elle ?',
  'Confidence factor': 'Facteur de confiance',
  'Confidence factor calculation': 'Calcul du facteur de confiance',
  'Do you actually enjoy doing the tasks of your job?':
    'Aimez-vous réellement faire les tâches de votre poste ?',
  'Do you appreciate the quality of the teamwork?':
    'Appréciez-vous la qualité du travail d’équipe ?',
  'Do you have time for your personal projects?':
    'Avez-vous du temps pour vos projets personnels ?',
  'Do you like your workplace?': 'Aimez-vous votre lieu de travail ?',
  'Do your career prospects suit you?':
    'Vos perspectives d’évolution professionnelles vous convient-elles ?',
  'Do your colleagues agree with you?':
    'Vos collègues sont-ils d’accord avec vous ?',
  'Does doing this work bring you closer to your career goals?':
    'Exercer ce travail vous rapproche-t-il de vos objectifs professionnels ?',
  'Is your company doing well?': 'Votre société se porte-t-elle bien ?',
  lang: 'fr',
  languageId: 'Français',
  Print: 'Impression',
  Result: 'Résultat',
  'Source : Brett Nelson': 'Source : Brett Nelson',
  'The job is for you.': 'Le job vous correspond.',
  'Think twice about taking the job, or leaving it right away.':
    'Songez à deux fois avant d’accepter le job, ou le quitter sur-le-champ.',
  'You got the perfect job!': 'Vous avez le job parfait !',
  "You're on the right track to finding the perfect job.":
    'Vous êtes sur la bonne voie pour trouver le job parfait.',
};

export default I18N_DICTIONARY;
