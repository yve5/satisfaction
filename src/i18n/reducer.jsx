import { i18nReducer } from 'lapwing';
import fr from './fr';
import en from './en';

export default i18nReducer([en, fr]);
