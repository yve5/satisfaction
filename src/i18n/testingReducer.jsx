import { i18nReducer } from 'organe';
import fr from './fr';
import en from './en';

export default i18nReducer([fr, en])();
