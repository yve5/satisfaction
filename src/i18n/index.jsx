import testingReducer from './testingReducer';

export default testingReducer;
