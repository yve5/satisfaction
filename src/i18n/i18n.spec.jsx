import React from 'react';
import configureMockStore from 'redux-mock-store';

import { thunk } from 'redux-thunk';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import i18n from '.';
import en from './en';
import fr from './fr';

import {
  I18n,
  trimTerms,
  checkMissingTerms,
  checkLocalTermOrder,
  convertLocalCollection,
} from 'organe';

describe('Local i18n', () => {
  it('should detect missing terms', () => {
    expect(checkMissingTerms(en, fr)).toEqual([]);
  });

  it('should check english term order', () => {
    expect(checkLocalTermOrder(en)).toEqual(convertLocalCollection(en));
  });

  it('should check french term order', () => {
    expect(checkLocalTermOrder(fr)).toEqual(convertLocalCollection(fr));
  });

  it('should trim local dictionaries', () => {
    expect(trimTerms(en)).toEqual([]);
  });

  it('should match expected snapshot', () => {
    const mockStore = configureMockStore([thunk]);

    const { asFragment } = render(
      <Provider store={mockStore({ i18n })}>
        {['Hello World'].map((term, index) => (
          <p key={index}>
            <I18n>{term}</I18n>
          </p>
        ))}
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
