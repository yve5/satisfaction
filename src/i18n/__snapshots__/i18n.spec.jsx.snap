// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Local i18n should match expected snapshot 1`] = `
<DocumentFragment>
  <p>
    Bonjour le monde
  </p>
</DocumentFragment>
`;
